packages/checkout-storefront/src/components/AddressForm/useAddressFormValidationResolver.ts

```diff
@@ -0,0 +1,67 @@
+import { AddressFormData } from "@/checkout-storefront/components/AddressForm/types";
+import { Errors, useErrorMessages } from "@/checkout-storefront/hooks";
+import { usePhoneNumberUtils } from "@/checkout-storefront/hooks/usePhoneNumberUtils";
+import { getErrorsAsObject, getAllValidationErrors } from "@/checkout-storefront/lib/utils";
+import { useCallback } from "react";
+import { object, string, ValidationError as YupValidationErrorObject } from "yup";
+
+interface ResolverResult {
+  errors: Errors<AddressFormData>;
+  values: Partial<AddressFormData>;
+}
+
+export const useAddressFormValidationResolver = () => {
+  const { errorMessages } = useErrorMessages();
+  const { isValidNumber } = usePhoneNumberUtils();
+
+  const schema = object({
+    firstName: string().required(errorMessages.required),
+    lastName: string().required(errorMessages.required),
+    streetAddress1: string().required(errorMessages.required),
+    streetAddress2: string(),
+    companyName: string(),
+    postalCode: string().required(errorMessages.required),
+    city: string().required(errorMessages.required),
+    cityArea: string(),
+    countryArea: string(),
+    phone: string().required(errorMessages.required),
+  });
+
+  return useCallback(
+    async (data: AddressFormData) => {
+      let result: ResolverResult = { errors: {}, values: {} };
+
+      try {
+        const values = await schema.validate(data, {
+          abortEarly: false,
+        });
+
+        result = {
+          ...result,
+          values,
+        };
+      } catch (error) {
+        const errors = getErrorsAsObject(getAllValidationErrors(error as YupValidationErrorObject));
+        result = { ...result, errors };
+      }
```

Comment:
I think we could extract it to a function `getAddressFormResolverValue` or something, and avoid using `let` then.

Reply:
tbh I added it like that knowing that it's like the resolvers will be refactored soon along with changing the forms lib so for now I'd just leave it as it is